internal/controllers/topology/cluster/desired_state_test.go

```diff
@@ -1603,6 +1606,105 @@ func TestComputeMachineDeployment(t *testing.T) {
 		// Check that UnhealthyConditions are set as expected.
 		g.Expect(actual.MachineHealthCheck.Spec.UnhealthyConditions).To(Equal(unhealthyConditions))
 	})
+	t.Run("Should correctly set the finalizer", func(t *testing.T) {
+		g := NewWithT(t)
+		s := scope.New(cluster)
+		s.Blueprint = blueprint
+		deletionTimestamp := metav1.Now()
+		currentMd := &clusterv1.MachineDeployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "existing-deployment-1",
+			},
+			Spec: clusterv1.MachineDeploymentSpec{
+				Replicas: pointer.Int32(3),
+				Template: clusterv1.MachineTemplateSpec{
+					Spec: clusterv1.MachineSpec{
+						Version: pointer.String("v1.21.2"),
+						Bootstrap: clusterv1.Bootstrap{
+							ConfigRef: contract.ObjToRef(workerBootstrapTemplate),
+						},
+						InfrastructureRef: *contract.ObjToRef(workerInfrastructureMachineTemplate),
+					},
+				},
+			},
+		}
+		mdTopology = clusterv1.MachineDeploymentTopology{
+			Metadata: clusterv1.ObjectMeta{
+				Labels: map[string]string{"foo": "baz"},
+			},
+			Class: "linux-worker",
+			Name:  "big-pool-of-machines",
+		}
+		t.Run("Should set the finalizer on a new MD", func(t *testing.T) {
+			s.Current.MachineDeployments = map[string]*scope.MachineDeploymentState{
```

Comment:
This looks like the case where the MD already exists? Does this name actually belong to the next test case and vice versa?